include/trompeloeil.hpp

```diff
@@ -529,10 +849,23 @@ namespace trompeloeil
   class duck_typed_matcher : public matcher
   {
   public:
+#if TROMPELOEIL_GCC && TROMPELOEIL_GCC_VERSION < 40900
```

Comment:
I'm a bit confused by the `operator std::string&&() const` added here. The comment implies that this is just to make the self test compile, but does this mean that duck typed matchers do not work with g++-4.8? If so, I wonder if it wouldn't be better to exclude those tests when building with g++-4.8.

Reply:
This is a long story to be told with whisky and tears in equal parts.  The code here is the product of many other failed experiments.  I am investigating the consequences of removing support for these `g++-4.8`-specific user-defined conversions in `duck_typed_matcher` and `wildcard` on my production code base and will have a better idea of the direction to take in the next day or so.

Reply:
We live about as far apart as we can on this planet, but should we meet, I'll bring the whisky and you can cry your eyes out ;-)
No worries.

Reply:
I disabled the non-template user-defined conversions  in `wildcard` and
`duck_typed_matcher` after finding
- No production code was affected by not having support for implicit
  conversions from `wildcard` or `duck_typed_matcher` to rvalue references.
- Only five test cases failed without these user-defined conversions.
  These are now guarded with `TROMPELOEIL_TEST_RVALUE_REFERENCE_FAILURES`.
  Interestingly only `wildcard` test cases failed: I could not find a test
  case failure with `duck_typed_matcher`.
  
Documentation in `Backward.md` has been updated to explain the shortcomings
of `g++-4.8` should someone stumble upon this issue.

It also appears that the impact on coverage as reported by Coveralls has
been minimal.